softaculous_plugin.php

```diff
@@ -55,21 +167,77 @@ public function softAutoInstall($event)
                 )
             );
 
+        if ($service_activated) {
+            $this->runInstaller($par['service_id']);
+        }
+    }
+
+    /**
+     * Run the installer for the given service and class
+     *
+     * @param int $service_id The ID of the service for which to run the script
+     * @throws Exception
+     */
+    private function runInstaller($service_id)
+    {
+        Loader::loadModels($this, ['Services', 'ModuleManager', 'Softaculous.SoftaculousQueuedServices']);
+
+        // Fetch necessary data
+        $service = $this->Services->get($service_id);
+        $module_row = $this->ModuleManager->getRow($service->module_row_id);
+
+        // Get module info
+        $module_info = $this->getModuleClassByPricingId($service->pricing_id);
+
         // This plugin only supports the follwing modules: cPanel, CentOS Web Panel, and Plesk
         $accepted_modules = ['cpanel', 'centoswebpanel', 'plesk'];
-        if ($service_activated && $module_info && in_array($module_info->class, $accepted_modules)) {
-            // Fetch necessary data
-            $service = $this->Services->get($par['service_id']);
-            $module_row = $this->ModuleManager->getRow($service->module_row_id);
-            try {
-                $installer = $this->loadInstaller($module_info->class);
-
-                $installer->install($service, $module_row->meta);
-            } catch (Throwable $e) {
-                throw new Exception(Language::_('SoftaculousPlugin.library_error', true));
-            } catch (Exception $e) {
-                throw new Exception(Language::_('SoftaculousPlugin.library_error', true));
+        if (!$module_info || !in_array($module_info->class, $accepted_modules)) {
+            // Remove the service from the queue since it is invalid
+            $this->SoftaculousQueuedServices->delete($service_id);
+            return;
+        }
+
+        try {
+            // Load the installer for this module
+            $installer = $this->loadInstaller($module_info->class);
+
+            // Run the installation script
+            $installer->install($service, $module_row->meta);
+            if (($errors = $installer->errors())) {
```

Comment:
Error logging is already done in each installer class, but I've added a log for success

Reply:
The logger property of the installer class is protected (as it should be), so it wouldn't be available for use here since this class doesn't extend the installer class. You would have to load it from the container instead.

Reply:
Updated